src/Spago/PackageSet.hs

```diff
@@ -220,6 +221,42 @@ localImportPath (Dhall.Import
 localImportPath _ = Nothing
 
 
+rootPackagePath :: Dhall.Import -> Maybe System.IO.FilePath
+rootPackagePath (Dhall.Import
+  { importHashed = Dhall.ImportHashed
+    { importType = localImport@(Dhall.Local _ Dhall.File { file = "packages.dhall" })
+    }
+  , Dhall.importMode = Dhall.Code
+  })              = Just $ Text.unpack $ pretty localImport
+rootPackagePath _ = Nothing
+
+
+-- | In a Monorepo we don't wish to rebuild our shared packages over and over, 
+-- | so we build into an output folder where our root packages.dhall lives
+findRootOutputPath :: Spago m => System.IO.FilePath -> m (Maybe System.IO.FilePath)
+findRootOutputPath path = do
+  echoDebug "Locating root path of packages.dhall"
+  imports <- liftIO $ Dhall.readImports $ Text.pack path
+  let localImports = catMaybes (map rootPackagePath imports)
```

Comment:
Can be simplified using [mapMaybe](https://www.stackage.org/haddock/lts-14.7/base-4.12.0.0/Data-Maybe.html#v:mapMaybe)
```suggestion
  let localImports = mapMaybe rootPackagePath imports
```